src/borg/testsuite/archiver.py

```diff
@@ -1298,9 +1298,12 @@ def test_list_size(self):
         self.cmd('init', self.repository_location)
         test_archive = self.repository_location + '::test'
         self.cmd('create', '-C', 'lz4', test_archive, 'input')
-        output = self.cmd('list', '--format', '{size} {csize} {path}{NL}', test_archive)
-        size, csize, path = output.split("\n")[1].split(" ")
+        output = self.cmd('list', '--format', '{size} {csize} {dsize} {dcsize} {path}{NL}', test_archive)
+        size, csize, dsize, dcsize, path = output.split("\n")[1].split(" ")
         assert int(csize) < int(size)
+        assert int(dcsize) < int(dsize)
+        assert int(dsize) <= int(size)
+        assert int(dcsize) <= int(csize)
```

Comment:
However, as you envisioned, the dsize value (calculated as original @Abogical (and mine) code) fails (--format "{dsize:>10} {size:>10} {path}{NL}")
```
   4194304  524288000 TEST2.img
```